tests/README.md

```diff
@@ -0,0 +1,515 @@
+values to be legitimate. In the case of these particular conditions, they are tested 
+[here](https://github.com/ethereum/consensus-specs/blob/926e5a3d722df973b9a12f12c015783de35cafa9/tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py#L87-L104).
+One test checks what happens if the attestation is too early, and another if it is too late.
+
+However, it is not enough to ensure we reject invalid blocks. It is also necessary to ensure we accept all valid blocks. You saw earlier
+a test (`test_success`) that tested that being `MIN_ATTESTATION_INCLUSION_DELAY` after the data for which we attest is enough. 
+Now we'll write a similar test that verifies that being `SLOTS_PER_EPOCH` away is still valid. To do this, we modify the 
+`test_after_epoch_slots` function. We need two changes:
+
+1. Call `transition_to_slot_via_block` with one less slot to advance
+1. Don't tell `run_attestation_processing` to return an empty post state.
+
+The modified function is:
+
+```python
+@with_all_phases
+@spec_state_test
+def test_almost_after_epoch_slots(spec, state):
+    attestation = get_valid_attestation(spec, state, signed=True)
+
+    # increment to latest inclusion slot (not beyond it)
+    transition_to_slot_via_block(spec, state, state.slot + spec.SLOTS_PER_EPOCH)
+
+    yield from run_attestation_processing(spec, state, attestation)
+```    
+
+Add this function to the file `consensus-specs/tests/core/pyspec/eth2spec/test/phase0/block_processing/test_process_attestation.py`,
+and run the test:
+
+```sh
+cd ~/consensus-specs
+. venv/bin/activate
+cd tests/core/pyspec
+python -m pytest -k almost_after --fork Merge eth2spec
+```
+
+You should see it ran successfully (although you might get a warning, you can ignore it)
+
+<!--
+
+https://ethos.dev/beacon-chain/
+
+-->
+
+## How are These Tests Used?
+
+So far we've ran tests against the formal specifications. This is a way to check the specifications
+are what we expect, but it doesn't actually check the beacon chain clients. The way these tests get applied
+by clients is that every few days 
+[new test specifications are released](https://github.com/ethereum/consensus-spec-tests/releases),
+in a format [documented here](https://github.com/ethereum/consensus-specs/tree/dev/tests/formats).
+All the clients know how to use tests in the format. 
+
+<!-- 
+### ZRNT, an Example
+
+To actually see the process end to end we are going to use [this client](https://github.com/protolambda/zrnt).
+
+1. ZRNT is written in Go, so the first step is to 
+   [install the Go programming language on your environment](https://golang.org/doc/install).
+1. Download ZRNT itself, and the current tests
+
+   ```sh
+   git clone https://github.com/protolambda/zrnt.git
+   cd zrnt
+   make download-tests
+   ```
+   
+1. Run the attestation tests
+
+   ```sh
+   cd tests/spec
+   go test -tags preset_minimal ./test_runners/operations/attestation_test.go 
```

Comment:
The `-tags preset_minimal ` build-tag isn't necessary anymore, ZRNT currently supports full runtime configuration of everything. I need to update my docs :sweat_smile:

Reply:
The ZRNT material is commented out. I decided that running a test in ZRNT isn't impressive enough to be worth doing, all you can see is that the test passed successfully. Or am I wrong, and there's a way for ZRNT to produce an interesting trace (similar to the vmtrace in the execution layer)?